src/client/api/json_parser/outputs/output_foundry.c

```diff
@@ -8,16 +8,150 @@
 #include "core/models/outputs/outputs.h"
 #include "core/utils/macros.h"
 
+/*
+"tokenScheme": {
+  "type" : 0,
+  "mintedTokens" : 200000000000000000000000000000000000000000,
+  "meltedTokens" : 100000000000000000000000000000000000000000,
+  "maximumSupply": "30000000000000000000000000000000000000000",
+}
+*/
+int json_token_scheme_deserialize(cJSON *output_obj, token_scheme_t **token_scheme) {
+  if (output_obj == NULL || *token_scheme != NULL) {
+    printf("[%s:%d]: Invalid parameters\n", __func__, __LINE__);
+    return -1;
+  }
+
+  // token scheme type
+  uint8_t scheme_type;
+  if (json_get_uint8(output_obj, JSON_KEY_TYPE, &scheme_type) != JSON_OK) {
+    printf("[%s:%d]: getting %s json uint8 failed\n", __func__, __LINE__, JSON_KEY_TYPE);
+    return -1;
+  }
+
+  if (scheme_type == SIMPLE_TOKEN_SCHEME) {
+    uint256_t *mintedt_tokens = NULL;
+    uint256_t *melted_tokens = NULL;
+    uint256_t *max_supply = NULL;
+    // minted tokens
+    char temp_str[STRING_NUMBER_MAX_CHARACTERS];
+    if (json_get_string(output_obj, JSON_KEY_MINTED_TOKENS, temp_str, STRING_NUMBER_MAX_CHARACTERS) != JSON_OK) {
+      printf("[%s:%d]: getting %s json string failed\n", __func__, __LINE__, JSON_KEY_MINTED_TOKENS);
+      return -1;
+    }
+    mintedt_tokens = uint256_from_str(temp_str);
+
+    // melted tokens
+    memset(temp_str, 0, STRING_NUMBER_MAX_CHARACTERS);
+    if (json_get_string(output_obj, JSON_KEY_MELTED_TOKENS, temp_str, STRING_NUMBER_MAX_CHARACTERS) != JSON_OK) {
+      printf("[%s:%d]: getting %s json string failed\n", __func__, __LINE__, JSON_KEY_MELTED_TOKENS);
+      return -1;
+    }
+    melted_tokens = uint256_from_str(temp_str);
+
+    // maximum supply
+    memset(temp_str, 0, STRING_NUMBER_MAX_CHARACTERS);
+    if (json_get_string(output_obj, JSON_KEY_MAX_SUPPLY, temp_str, STRING_NUMBER_MAX_CHARACTERS) != JSON_OK) {
+      printf("[%s:%d]: getting %s json string failed\n", __func__, __LINE__, JSON_KEY_MAX_SUPPLY);
+      return -1;
+    }
+    max_supply = uint256_from_str(temp_str);
+
+    *token_scheme = token_scheme_simple_new(mintedt_tokens, melted_tokens, max_supply);
+    if (!token_scheme) {
+      printf("[%s:%d]: creating token scheme object failed\n", __func__, __LINE__);
+      return -1;
+    }
```

Comment:
`mintedt_tokens`, `melted_tokens` and `max_supply` needs to be freed.

Reply:
Done.